core/src/serve_repair.rs

```diff
@@ -516,6 +524,18 @@ impl ServeRepair {
             inc_new_counter_debug!("serve_repair-handle-repair--eq", stats.self_repair);
         }
 
+        let staked_nodes = self.bank_forks.read().unwrap().root_bank().staked_nodes();
```

Comment:
This will hold the lock on `bank_forks` until `staked_nodes` returns:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=493a414a85328d8410c4de52d54c85fa

`staked_nodes` is lazily computed:
https://github.com/solana-labs/solana/blob/836ba734e/runtime/src/vote_account.rs#L91-L108

so this potentially may add lock contention on bank-forks. To avoid that: 
```rust
let root_bank = self.bank_forks.read().unwrap().root_bank(); // drops the lock on bank_forks
let staked_nodes = root_bank.staked_nodes();
```

But I think calculating staked-nodes for metrics purposes is very wasteful.
Might instead just use `epoch_stakes`.

Reply:
Won't the `staked_nodes` calculation also be performed by `ClusterInfo::run_listen`? This metrics calculation is only performed every 3 seconds.

regardless, updated to use `epoch_stakes`.

Reply:
> Won't the staked_nodes calculation also be performed by ClusterInfo::run_listen?

Probably `ClusterInfo::run_listen` should also use `epoch_staked_nodes` instead.